src/App.jsx

```diff
@@ -14,45 +16,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const REVERSE_DEFAULT_VALUE = false;
+const SORT_FIELD_ALPHABETICAL = 'alphabetical';
+const SORT_FIELD_LENGTH = 'length';
+
+function getGoodsPrepared(goods, sortField, reverseSort) {
+  const goodsPrepared = [...goods];
+
+  if (sortField) {
+    goodsPrepared.sort((a, b) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABETICAL:
+          return a.localeCompare(b);
+
+        case SORT_FIELD_LENGTH:
+          return a.length - b.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseSort) {
+    goodsPrepared.reverse();
+  }
+
+  return goodsPrepared;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+function renderButton(label, onClick, isActive, colorClass) {
```

Comment:
This function effectively is a function component - so I would make it a component (and move to separate file)

```suggestion
const FancyButton = ({label, onClick, isActive, colorClass}) => {
```